frontend/test/metabase/lib/formatting/filesize-integration.unit.spec.js

```diff
@@ -0,0 +1,161 @@
+          number_style: "filesize",
+        }),
+      }));
+
+      expect(formattedData[0].file_size_formatted).toBe("1 KiB");
+      expect(formattedData[1].file_size_formatted).toBe("5 MiB");
+      expect(formattedData[2].file_size_formatted).toBe("1 GiB");
+    });
+
+    it("should handle unit in header setting", () => {
+      const value = 1536;
+
+      // Unit in cell
+      const inCell = formatNumber(value, {
+        number_style: "filesize",
+        filesize_unit_in_header: false,
+        type: "cell",
+      });
+      expect(inCell).toBe("1.5 KiB");
+
+      // Unit in header
+      const inHeader = formatNumber(value, {
+        number_style: "filesize",
+        filesize_unit_in_header: true,
+        type: "cell",
+      });
+      expect(inHeader).toBe("1.5");
+    });
+  });
+
+  describe("Chart Visualizations", () => {
+    it("should format Y-axis values for bar charts", () => {
+      const yAxisValues = [0, 1000000, 2000000, 3000000, 4000000, 5000000];
+
+      // Test with decimal units for cleaner chart labels
+      const formattedValues = yAxisValues.map((val) =>
+        formatNumber(val, {
+          number_style: "filesize",
+          filesize_unit_system: "decimal",
+        }),
+      );
+
+      expect(formattedValues).toEqual([
+        "0 B",
+        "1 MB",
+        "2 MB",
+        "3 MB",
+        "4 MB",
+        "5 MB",
+      ]);
+    });
+
+    it("should handle large values in line charts", () => {
+      const dataPoints = [
+        1099511627776, // 1 TiB
+        2199023255552, // 2 TiB
+        5497558138880, // 5 TiB
+        10995116277760, // 10 TiB
+      ];
+
+      const formatted = dataPoints.map((val) =>
+        formatNumber(val, { number_style: "filesize" }),
+      );
+
+      expect(formatted).toEqual(["1 TiB", "2 TiB", "5 TiB", "10 TiB"]);
+    });
+  });
+
+  describe("Export Functionality", () => {
+    it("should maintain formatting in CSV exports", () => {
+      // Simulating data that would be exported
+      const exportData = {
+        bandwidth_used: 1073741824,
+        data_transferred: 5368709120,
+        cache_size: 268435456,
+      };
+
+      const csvFormatted = Object.entries(exportData).reduce(
+        (acc, [key, value]) => {
+          acc[key] = formatNumber(value, { number_style: "filesize" });
+          return acc;
+        },
+        {},
+      );
+
+      expect(csvFormatted).toEqual({
+        bandwidth_used: "1 GiB",
+        data_transferred: "5 GiB",
+        cache_size: "256 MiB",
+      });
+    });
+  });
+
+  describe("Auto-detection", () => {
+    it("should auto-detect file size columns", () => {
+      const columns = [
+        { name: "id" },
+        { name: "file_size" },
+        { name: "bandwidth" },
+        { name: "bytes_transferred" },
+        { name: "revenue" },
+      ];
+
+      // This simulates what getDefaultNumberStyle would do
+      const shouldUseFileSize = (column) => {
```

Comment:
this test tests function that is defined in the test

Reply:
Yup, good spot. Will correct